operator/cmd/root.go

```diff
@@ -106,8 +106,15 @@ var (
 			}
 		}),
 
+		cell.Provide(func(operatorCfg *operatorOption.OperatorConfig) client.SharedConfig {
+			return client.SharedConfig{
+				SkipCRDCreation: operatorCfg.SkipCRDCreation,
+			}
```

Comment:
I think we can get rid of the "SharedConfig" trick. Usually we want that to break the direct dependency of a cell config to global options.

But in this case, the `SkipCRDCreation` option is only used by the new cell, so it is not "global" anymore.
Thus we can remove that option from the `operatorOption.OperatorConfig` struct (alongside any reference to it in `operator/cmd/fags.go`, `operator/option/config.go` and `pkg/option/config.go`) and let the cell declare and initialize it through `cell.Config`.

Something similar can be seen [here](https://github.com/cilium/cilium/blob/master/clustermesh-apiserver/health.go#L19-L33) for the `ClusterMeshHealthPort` option (even though here the declaration of the option name and its default value have been kept in packages `option` and `defaults` respectively).

Please note that this will break the controlplane test setup in `test/controlplane/testcase.go`, where we force-skip the CRDs registration setting `operatorOption.Config.SkipCRDCreation = true` in `*ControlPlaneTest.SetupEnvironment()`.

To fix this you can add something like this:

```
   vp := cpt.operatorHandle.hive.Viper()
   vp.Set(client.SkipCRDCreation, true)
```

in `*ControlPlaneTest.StartOperator()`, just after the operator hive instantiation.

Reply:
Really appreciate the detailed guidance 👍 I have updated the PR, please take a second look